src/object-pool.ts

```diff
@@ -1,10 +1,287 @@
-import {ArmorObjectPoolOptions} from './object-pool-options';
-import {ArmorObjectPoolState} from './object-pool-state';
+import ArmorCollection from './collection';
+import ArmorCollectionSelector from './selector';
+import ArmorObjectPoolInstance from './object-pool-instance';
+import ArmorObjectPoolOptions from './object-pool-options';
+import ArmorObjectPoolState from './object-pool-state';
 
-export class ArmorObjectPool<T> {
-	public readonly state: ArmorObjectPoolState;
+export default class ArmorObjectPool<T> implements ArmorCollection<T> {
+	public state: ArmorObjectPoolState<T>;
+	public startSize: number = 10;
+	public readonly poolObj: ArmorObjectPoolInstance<T>;
 
-	constructor(options?: ArmorObjectPoolOptions) {
-		this.state = new ArmorObjectPoolState(options);
+	constructor(poolObj: ArmorObjectPoolInstance<T>, options?: ArmorObjectPoolOptions<T>) {
+		if (typeof poolObj !== 'function') {
+			throw new Error('Must have a comparator function for priority queue to operate properly');
+		}
+		this.poolObj = poolObj;
+
+		this.state = {
+			type: 'opState',
+			elements: [],
+			objectCount: 0,
+			maxSize: 1000,
+			autoIncrease: false,
+			increaseFactor: 2,
+			increaseBreakPoint: 0.8
+		};
+
+		if (options) {
+			this.parseOptions(options);
+		}
+
+		this.increase(this.startSize);
+	}
+	public parseOptions(options: ArmorObjectPoolOptions<T>): void {
```

Comment:
Here's how the changes would look here:
```typescript
public parseOptions(options?: ArmorObjectPoolOptions<T>): ArmorObjectPoolState<T> {
   const state = {
      startSize: this.parseOptionsStartSize(options),
      ...
   }
}
```
The above is an approximation but the rough pattern. Create default hard-coded values like `const state = ...`, then update it's properties as you call each parse function. Note above I passed options to `parseOptionsStartSize` because `options.startSize` will throw if options is undefined. I can't early drop on `!options` here though until I actually defined my starting object since it must be returned. You may find yourself passing the current state and options to each parse function.